tests/oauth2/test_reauth.py

```diff
@@ -326,6 +326,26 @@ def test_refresh_grant_failed(mock_metrics_header_value):
         )
 
 
+def test_refresh_grant_failed_with_string_type_response():
+    with mock.patch(
+        "google.oauth2._client._token_endpoint_request_no_throw"
+    ) as mock_token_request:
+        mock_token_request.return_value = (False, "string type error", False)
+        with pytest.raises(exceptions.RefreshError) as excinfo:
+            reauth.refresh_grant(
+                MOCK_REQUEST,
+                "token_uri",
+                "refresh_token",
+                "client_id",
+                "client_secret",
+                scopes=["foo", "bar"],
+                rapt_token="rapt_token",
+                enable_reauth_refresh=True,
```

Comment:
in continuation from above comment.
In other words, i would like to understand what should be the expected behaviour of the method for the below test case?

`    with mock.patch(
        "google.oauth2._client._token_endpoint_request_no_throw"
    ) as mock_token_request:
        mock_token_request.return_value = (False, _REAUTH_NEEDED_ERROR, False)
        with pytest.raises(exceptions.RefreshError) as excinfo:
            reauth.refresh_grant(
                MOCK_REQUEST,
                "token_uri",
                "refresh_token",
                "client_id",
                "client_secret",
                scopes=["foo", "bar"],
                rapt_token="rapt_token",
                enable_reauth_refresh=True,
           )
           `
whether RefreshError exception be expected or whether reauth refresh flow (i.e., 354-364) is expected since enable_reauth_refresh=True and reponse_data=_REAUTH_NEEDED_ERROR?

Reply:
@clundin25 , please let me know your thoughts on this.

Reply:
I believe in case of reauth needed error, the oauth2 token endpoint always returns a json, in other words, something like `mock_token_request.return_value = (False, {"error":_REAUTH_NEEDED_ERROR, ...}, False)`. The server won't return `mock_token_request.return_value = (False, _REAUTH_NEEDED_ERROR, False)` as described in the test case you mentioned.

Reply:
ok. then agree.